File: JavaSource/src/main/java/ru/akhcheck/patterns/visitor/workers/Windower.java
package ru.akhcheck.patterns.visitor.workers;

import ru.akhcheck.patterns.visitor.rooms.BathRoom;
import ru.akhcheck.patterns.visitor.rooms.Home;
import ru.akhcheck.patterns.visitor.rooms.Kitchen;
import ru.akhcheck.patterns.visitor.rooms.LivingRoom;

/** Оконщик: ставит окна там, где они бывают. */
public class Windower implements Visitor {
    @Override
    public void visit(Kitchen kitchen) {
        kitchen.getWall().setMaterial("brick with window");
    }

    @Override
    public void visit(LivingRoom livingRoom) {
        livingRoom.getWall().setMaterial("brick with panoramic window");
    }

    @Override
    public void visit(BathRoom bathRoom) {
        // В санузле окна не ставим.
    }

    @Override
    public void visit(Home home) {
        System.out.println("Windower came to the house");
    }
}

File: Visitor/python-source/workers/Windower.py
from __future__ import annotations

from typing import TYPE_CHECKING

from .Visitor import Visitor

if TYPE_CHECKING:
    from rooms import BathRoom
    from rooms import Home
    from rooms import Kitchen
    from rooms import LivingRoom


class Windower(Visitor):
    """Оконщик: ставит окна там, где они бывают."""

    def visit_kitchen(self, kitchen: Kitchen) -> None:
        kitchen.wall.material = "brick with window"

    def visit_living_room(self, living_room: LivingRoom) -> None:
        living_room.wall.material = "brick with panoramic window"

    def visit_bath_room(self, bath_room: BathRoom) -> None:
        """В санузле окна не ставим."""

    def visit_home(self, home: Home) -> None:
        print("Windower came to the house")
